### ISSUE TITLE
Indians offering worthless\annoying deals

### ISSUE THREAD

**Issue by ErikSandbraaten** [We-the-People-civ4col-mod/Mod#131]
Indians should stop offering worthless trades late in the game (like 6 units of tobacco)

If you refuse, there's a chance to anger the natives. Additionally, this features causes a slight lag and it's often annoying to have unnecessary popups at the start of the turn when there so much other stuff going on that is far more important.

**REFERENCES**

See:

CvPlayerAI::AI_nativeTrade(CvUnit* pUnit) where the yield is selected.
DIPLOEVENT_NATIVE_TRADE where the choice made by the player i handled

 CvUnitAI::AI_defensiveBraveMove() .. if (AI_getUnitAIState() == UNITAI_STATE_BEARING_GIFTS)..
where the native unit chooses the city to target for the trade.
CvUnitAI::AI_defensiveBraveMove that sets AI_setUnitAIState(UNITAI_STATE_BEARING_TRADE)
This controls if a brave will be carrying goods for trade.

The latter is the real problem since in the late game due to the number of braves, it’s possible for this to happen too frequently. We could scale it with the game turn. The same could be done for gifts.

**Comment by NoFreeUsernameDammit** [comment]
I agree. In my experience, by mid/late game, natives either offer small and annoying trades/gifts that are not worth the bother of transporting and utilizing since they're so irregular, or tend to dump on you huge amount of goods at once (I've seen offers of over 600 units on Marathon) that they create logistical issue.

I'd prefer the gifts and trades diminish and disappear by mid/late game and were replaced by some kind of trade post system that would allow more reliable regular trade with natives.

**Comment by abmpicoli** [comment]
I think they should offer big stacks , not small quantities per time. And they should be like a trade offer, like we have with them... where we could haggle for prices... 

I can't put a finger on it, but I think the natives start trading stuff when they are nearly overflowing their own stocks... So the algorithm could be changed to generate big stacks at a time, instead of small amounts. 

And I totally agree that we should have a kind of agreement, where the natives agree to sell a specific amount of goods at a time... And then they would deliver it to us at regular intervals... These agreements would generate a diplomatic bonus with them... And would avoid the random flow to an extent...

**Comment by raystuttgart** [comment]
I basically implemented this feature "Natives coming to your Villages to sell tiny amounts of Goods" just for adding **a bit of flavour / atmosphere**.
It was never meant to replace selling big amouts of Goods by Wagon Trains or Ships to them.

The logic when they choose to send a Unit to you to trade is more or less dependend on some random and on Attitude.

I never thought about scaling it thoug in any way to Turns passed. :dunno:
If there is a good suggestion, let me know. I could improve it.

**Comment by Nightinggale** [comment]
Dealing with natives is kind of like a culture clash. The Europeans could buy expensive stuff for nearly nothing, but at the same time the natives viewing worthless stuff as important could very well be offended if the Europeans refused what the Europeans viewed as worthless gifts.

Being diplomatic and gaining a good relationship with the natives shouldn't come free. Filling up colonies with such gifts or trade results is something, which happens if you want a good relationship.

Also the annoying part of it could be fixed by implementing #144 as it would avoid the issue that 6 tobacco takes up an entire cargo slot and leave 70 cotton behind.

**Comment by raystuttgart** [comment]
For now I will not work on it.

I currently just don't have a good idea what to do with it that would create a result I am happy with. 
(I don't want it to become too rare in endgame and I don't want it to become overpowered in endgame either.)

It is basically meant to be a "flavour / atmosphere" event mainly.
